pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,210 @@
+	_, err := pc.isNodeInEndpoints()
+	if err != nil {
+		// Add node if not present in the endpoints list
+		patchData := fmt.Sprintf("[{\"op\": \"add\", \"path\": \"/subsets/0/addresses/-\", \"value\":"+
+			" {\"ip\": \"%s\", \"targetRef\": {\"kind\": \"Node\", \"name\":\"%s\"}}}]",
+			pc.endpointAddress.IP, pc.hostname)
+
+		_, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Patch(context.TODO(),
+			windowsMetricsEndpoints, types.JSONPatchType, []byte(patchData), metav1.PatchOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "could not add node %v to endpoints object, %v", pc.hostname, pc.endpointAddress.IP)
+		}
+		log.Info("prometheus configured", "operation", "add", "IPAddress", pc.endpointAddress.IP)
+		return nil
+	}
+	return nil
+}
+
+// removeEndpointAddress removes node ip address to the list of subsets in Endpoints object
+func (pc *prometheusNodeConfig) removeEndpointAddress() error {
+	nodeIPPosition, err := pc.isNodeInEndpoints()
+	// If node not found in the endpoints list, return without proceeding further
+	if err != nil {
+		return nil
+	}
+	patchData := fmt.Sprintf("[{\"op\": \"remove\", \"path\": \"/subsets/0/addresses/%s\"}]", nodeIPPosition)
+
+	_, err = pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Patch(context.TODO(),
+		windowsMetricsEndpoints, types.JSONPatchType, []byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "could not remove node %v from endpoints object", pc.hostname)
+	}
+	log.Info("prometheus configured", "operation", "remove", "IPAddress", pc.endpointAddress.IP)
+	return nil
+}
+
+// Add will create the Services and Service Monitors to allow the operator export the metrics by using
+// the Prometheus operator
+func Add(ctx context.Context, cfg *rest.Config, namespace string) {
+	// Add to the below struct any other metrics ports you want to expose.
+	servicePorts := []v1.ServicePort{
+		{Port: Port, Name: metricsPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: Port}},
+	}
+
+	// Create Service object to expose the metrics port(s).
+	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	if err != nil {
+		log.Error(err, "could not create metrics Service")
+	}
+	oclient, err := k8sclient.NewForConfig(cfg)
+	if err != nil {
+		log.Error(err, "could not create config clientset")
+	}
+
+	// When a selector is present in a headless service i.e. spec.ClusterIP=None, Kubernetes manages the
+	// list of endpoints reverting all the changes made by the operator. Remove selector from Metrics Service to avoid
+	// reverting changes in the Endpoints object.
+	patchData := fmt.Sprintf(`{"spec":{"selector": null }}`)
+	service, err = oclient.CoreV1().Services(namespace).Patch(ctx, service.Name, types.MergePatchType,
+		[]byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		log.Error(err, "could not remove selector from metrics service")
+	}
```

Comment:
Do we need to this if metrics is not enabled? Maybe you should create the `ServiceMonitor` first and only if that succeeds proceed to do this?